### DISCUSSION TITLE
Adding Modbus to Kura docker

### DISCUSSION THREAD

**Discussion by yevgenym1** [eclipse-kura/kura#5188] [General]
A newby here. I have launched latest Kura on Docker (Windows host) as described in the docs. I have added "Modbus Protocol Implementation for Eclipse Kura 4/5" package from the Marketplace, but I can't see anything under Driver and Assets / New Driver / Driver Factory. Any advise what to check? No error in the log. I can add and see S7 driver factory there.

**Comment by yevgenym1** [comment]
I have added ESF "Modbus Master Driver v2" driver as well (com.eurotech.modbus.driver) and only now I can see ord.eclipse.kura.driver.modbus in Driver Factory dropbox. Was it designed to be like that?

**Reply by mattdibi** [reply]
Hello there @yevgenym1, thanks for reaching out.

Yeah the plugin is working as designed. The "Modbus Protocol Implementation for Eclipse Kura 4/5" plugin adds the protocol implementation allowing developers to create applications that leverage the protocol. The "Modbus Master Driver v2" plugin instead adds the actual driver. I know it can be confusing...

One final remark: keep in mind that ESF is a *commercial* product derived from Kura and therefore is under a different license.

**Comment by mattdibi** [comment]
Feel free to reopen this if you have any additional questions!

**Reply by yevgenym1** [reply]
Does it mean that one cannot get free "off the shelf" Kura with Modbus driver for commercial usage (same as BACnet)?
But OPC UA and S7 are available for free in Kura for commercal usage?

**Reply by mattdibi** [reply]
To answer these questions I invite you to take a look at the [Modbus Driver plugin licensing](https://marketplace.eclipse.org/content/esf-modbus-driver) and the [ESF EULA](https://www.eurotech.com/download/en/portal/ESF+Eula)
![image](https://github.com/eclipse/kura/assets/22748355/0e411f6f-8d22-4da2-981c-96e3194d1b1c)

The same applies for all the packages you cited.
